ffi/capi/src/neo_datetime.rs

```diff
@@ -506,6 +479,292 @@ pub mod ffi {
             )))
         }
 
+        #[diplomat::rust_link(icu::datetime::fieldsets::zone::GenericShort, Struct)]
+        #[cfg(feature = "compiled_data")]
+        pub fn with_zone_generic_short(
+            &self,
+            locale: &Locale,
+        ) -> Result<Box<NeoZonedDateTimeFormatter>, DateTimeFormatterLoadError> {
+            formatter_with_zone(
+                &self.0,
+                locale,
+                icu_datetime::fieldsets::zone::GenericShort,
+                |names| {
+                    // NOTE: Keep this in sync with RawDateTimeNames::load_for_pattern
+                    names.as_mut().include_time_zone_essentials()?;
+                    names.as_mut().include_time_zone_generic_short_names()?;
+                    names.as_mut().include_time_zone_location_names()?;
```

Comment:
this is not great. can you add `include_for_field` on names, that takes a `(FieldSymbol, FieldLength)`?

Reply:
also not super happy about this. unsure of fix, ultimately okay with this landing especially with tests

Reply:
I know Shane was considering autogenerating some of this with askama/tera

Reply:
No. Each of these functions needs to be called on its own in order for data slicing to work. That's the whole point of having these separate constructors.

Reply:
could it be simplified with an include_for_field macro?

Reply:
> could it be simplified with an include_for_field macro?

A macro for this file or a macro exported by `icu_datetime`?

@robertbastian is not generally a fan of these types of macros (and I am also a bit reluctant to use them even though I use them a lot).

Reply:
I'm also happy to consider exporting something like `FixedCalendarDateTimeNames::include_time_zone_generic_short_names_with_fallback()` which calls all three functions. Then I could get rid of the "NOTE" comment. Would that help?

Reply:
https://github.com/unicode-org/icu4x/issues/5940#issuecomment-2722698945

Reply:
I think this should be `DateTimeNames::include_zone_generic_short_names`, where `zone_generic_short` matches `fieldsets::zone::GenericShort`. The fact that generic short timezone names are split across three data structs for deduplication reasons is not something a user of `DateTimeNames` should need to know, and the methods that load single data structs should be removed.